osu.Game/Overlays/Settings/Sections/Maintenance/MigrationSelectScreen.cs

```diff
@@ -4,109 +4,30 @@
 using System;
 using System.IO;
 using osu.Framework.Allocation;
-using osu.Framework.Graphics;
-using osu.Framework.Graphics.Containers;
-using osu.Framework.Graphics.Shapes;
 using osu.Framework.Logging;
 using osu.Framework.Platform;
 using osu.Framework.Screens;
 using osu.Game.Graphics;
 using osu.Game.Graphics.Sprites;
-using osu.Game.Graphics.UserInterface;
-using osu.Game.Graphics.UserInterfaceV2;
-using osu.Game.Screens;
-using osuTK;
 
 namespace osu.Game.Overlays.Settings.Sections.Maintenance
 {
-    public class MigrationSelectScreen : OsuScreen
+    public class MigrationSelectScreen : DirectorySelectScreen
     {
-        private DirectorySelector directorySelector;
+        [Resolved]
+        private Storage storage { get; set; }
 
-        public override bool AllowExternalScreenChange => false;
+        protected override DirectoryInfo InitialPath => new DirectoryInfo(storage.GetFullPath(string.Empty)).Parent;
 
-        public override bool DisallowExternalBeatmapRulesetChanges => true;
-
-        public override bool HideOverlaysOnEnter => true;
-
-        [BackgroundDependencyLoader(true)]
-        private void load(OsuGame game, Storage storage, OsuColour colours)
-        {
-            game?.Toolbar.Hide();
```

Comment:
This behaviour of hiding the toolbar has been removed, and I'm not seeing an explanation for it.

Reply:
The `InitialOverlayActivationMode` (`Disabled`) of the `DirectorySelectScreen` should cover this instead

Reply:
Ah, good point. Still not sure about where that is placed but that's relevant to one of the other comments.

Reply:
Okay, after the latest round of changes this *will* be broken again if I'm not mistaken, since `InitialOverlayActivationMode` is gone from the diff and the overrides have returned in their previous form, so the toolbar hide probably should too.

Reply:
I ended up moving the overrides specific to the migration select screen to the screen itself (which also includes hiding the toolbar upon entering, which is handled this time with the `HideOverlaysOnEnter` property.

Reply:
Oh, so that toolbar hide was just redundant the whole time? Alright then...